symDiskCache.py

```diff
@@ -0,0 +1,97 @@
+from symUtil import mkdir_p
+import cPickle as pickle
+import os
+from symLogging import LogDebug
+from collections import deque
+
+# Manage in disk symbol cache
+class SymDiskCache(object):
+  def __init__(self, cachePath, maxNumberOfFiles):
+    mkdir_p(cachePath)
+    self.cachePath = cachePath
+    self.maxNumberOfFiles = maxNumberOfFiles
+
+    self.entryCount, fileList = self._Walk(cachePath)
+
+    # sort the symbol relative to the last access time
+    fileList.sort(cmp=lambda a,b: cmp(a[1], b[1]))
+
+    # filter out access times
+    self.lru = deque(map(lambda x: x[0], fileList))
+
+    self.CheckCacheSize()
+
+  def Fetch(self, filename):
+    r"fetch a symbol file from the cache"
+    path = self.absPath(filename)
+    symbolInfo = None
+    if os.path.exists(path):
+      with open(path, 'r') as f:
```

Comment:
From https://docs.python.org/2/library/pickle.html

"Note: Be sure to always open pickle files created with protocols >= 1 in binary mode."